src/pybind/mgr/object_format.py

```diff
@@ -0,0 +1,272 @@
+# object_format.py provides types and functions for working with
+# requested output formats such as JSON, YAML, etc.
+
+import enum
+import errno
+import json
+
+from functools import wraps
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+try:
+    from typing import Protocol  # Protocol was added in Python 3.8
+except ImportError:
+
+    class Protocol:  # type: ignore
+        pass
+
+
+import yaml
+
+from mgr_module import HandlerFuncType
+
+
+class Format(enum.Enum):
```

Comment:
That will automatically implement the `__str__` for you (in case you need to serialize this enum):
```suggestion
class Format(str, enum.Enum):
```

Reply:
Sure. This was just a move from the older location. Otherwise I didn't change this type. But this would be a nice change....
I guess I did misunderstand you before about mixing structural and logical/behavioral changes in a PR - and I now assume you're OK with that.